source/includes/steps-fle-convert-to-a-remote-master-key-gcp.yaml

```diff
@@ -1,50 +1,91 @@
-  when using a KMS provider:
+  on a KMS provider:
 
   .. image:: /figures/CSFLE_Master_Key_KMS.png
      :alt: Diagram that describes creating a master key when using a KMS provider
 
-  In AWS management console, create a new symmetric master key in the KMS
-  section. Choose a name and description that helps you identify it; these
-  fields do not affect the functionality or configuration.
+  Create a **symmetric key** using the official Google Cloud
+  `Creating symmetric keys guide <https://cloud.google.com/kms/docs/creating-keys>`__.
+  This is your master key which you can access from your application.
+
+  .. note::
+
+     If you rotate the key whether by specifying a date or manually, the
+     provider changes the content of the master key. In order to transition
+     to using the new key, you need to perform the following steps:
 
-  In the :guilabel:`Usage Permissions` step of the key generation
-  process, add the full KMS ``List`` and ``Read`` permissions to the IAM
-  user you created in the previous step. This authorizes the user to encrypt
-  and decrypt the new master key.
+     1. decrypt and save the data using the original data encryption key
+     #. generate a new data encryption key with the new master key
+     #. update the ``keyVersion`` parameter to the latest version of the master key
+     #. re-encrypt the data using the new data encryption key
+
+  Make sure your application has appropriate roles to encrypt and decrypt
+  using the key. For example, the
+  `roles/cloudkms.cryptoKeyEncrypterDecrypter <https://cloud.google.com/kms/docs/reference/permissions-and-roles>`__
+  role has permissions to both encrypt and decrypt keys.
 
   .. important::
 
-     The new client IAM User *should not* have administrative permissions
-     for the master key.
+     The client application *should not* have administrative permissions
+     for the master key. We recommend that you follow the
+     `principle of least privilege <https://en.wikipedia.org/wiki/Principle_of_least_privilege>`__
+     to keep your data secure.
+
 ---
-title: Specify the AWS KMS Provider Credentials
-ref: specify-the-aws-kms-provider-credentials
+title: Specify your Google Cloud KMS Credentials
+ref: specify-your-gcp-kms-provider-credentials
 content: |
-  Unlike the local key provider, the AWS KMS provider does not read
+  Unlike the local key provider, the Google Cloud KMS does not read
   the master key directly from the client application. Instead,
-  it accepts the :guilabel:`Access Key ID` and :guilabel:`Secret Access
-  Key` configurations that point to the master key. The IAM user must have
-  the permissions set up in the previous step in order for the client to
-  use the KMS to encrypt and decrypt data encryption keys.
+  it retrieves the key using your service client credentials and key
+  details.
+
+  Configure your client application with the following authentication
```

Comment:
Suggestion: This feels like we need substeps, since we want them to do two things in order. Something like:

a. Configure your client application with the following authentication credentials.
table
b. Update the KMS Provider configuration in your CSFLE-enabled client
  creation code:
tabs

Reply:
Thanks, this makes sense. I'm going to move this suggestion to a new issue since it applies to content in multiple places.

Reply:
https://jira.mongodb.org/browse/DOCSP-13089

Reply:
https://jira.mongodb.org/browse/DOCSP-13089